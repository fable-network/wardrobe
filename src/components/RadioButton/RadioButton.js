import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import noop from '../../helpers/noop';

const Wrapper = styled('label')`
  position: relative;
  display: inline-flex;
  margin: 0;
  align-items: center;
  ${p => p.disabled && `opacity: 0.5;`};
`;

const StyledRadioInput = styled('div')`
  width: 0.776em;
  height: 0.776em;
  border-radius: 50%;
  box-shadow: 0px 0px 0px 0.05em ${p => p.theme.grey03};
  background: transparent;
  border: solid 0.05em #fff;
  outline: none;
`;

const Label = styled('span')`
  margin-left: 0.5em;
`;

const RadioInput = styled('input').attrs({ type: 'radio' })`
  opacity: 0;
  left: 0;
  position: absolute;
  cursor: ${p => (p.disabled ? 'not-allowed' : 'pointer')};
  width: 1em;
  height: 1em;
  &:checked {
    + ${StyledRadioInput} {
      box-shadow: 0px 0px 0px 0.05em ${p => p.theme.primary};
      background: ${p => p.theme.primary};
    }
  }
  &:focus {
    + ${StyledRadioInput} {
      box-shadow: 0 0 0.3em ${p => p.theme.primaryActive};
    }
  }
`;

const RadioButton = ({ label, selected, onSelect, name, ...props }) => (
  <Wrapper disabled={props.disabled}>
    <div>
      <RadioInput {...props} name={name} checked={selected} onClick={onSelect} />
      <StyledRadioInput />
    </div>
    {label && <Label>{label}</Label>}
  </Wrapper>
);

RadioButton.propTypes = {
  label: PropTypes.string,
  selected: PropTypes.bool,
  disabled: PropTypes.bool,
  onSelect: PropTypes.func,
  name: PropTypes.string,
};

RadioButton.defaultProps = {
  onSelect: noop,
};

export default RadioButton;
