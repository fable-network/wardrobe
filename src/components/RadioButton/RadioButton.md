Default (Uncontrolled):

```jsx
<RadioButton label="Radio label" />
```

Selected (Controlled):

```jsx
<RadioButton selected label="Radio label" />
```

Not selected (Controlled):

```jsx
<RadioButton selected={false} label="Radio label" />
```

Disabled:

```jsx
<RadioButton disabled label="Radio label" />
```

Radio button group (linked by the `name` prop):

```jsx
class Example extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = { selectedOption: null };
    this.handleSelect = this.handleSelect.bind(this);
  }

  handleSelect(e) {
    this.setState({ selectedOption: e.target.value });
  }

  render() {
    return (
      <>
        <p>
          <strong>Select an option:</strong>
        </p>
        <div>
          <RadioButton label="Option 1" name="someOption" value="1" onSelect={this.handleSelect} />
        </div>
        <div>
          <RadioButton label="Option 2" name="someOption" value="2" onSelect={this.handleSelect} />
        </div>
        <div>
          <RadioButton label="Option 3" name="someOption" value="3" onSelect={this.handleSelect} />
        </div>
        <pre>Selected Option: "{this.state.selectedOption}"</pre>
      </>
    );
  }
}
<Example />;
```
