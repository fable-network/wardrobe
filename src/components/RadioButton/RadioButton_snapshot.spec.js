/* eslint-disable */
// This file was automatically generated
import React from 'react';
import { shallow } from 'enzyme';
import RadioButton from './RadioButton';

describe('RadioButton Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
      const wrapper1 = shallow(<RadioButton label="Radio label" />);
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = shallow(<RadioButton selected label="Radio label" />);
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = shallow(<RadioButton selected={false} label="Radio label" />);
      expect(wrapper3).toMatchSnapshot();

      const wrapper4 = shallow(<RadioButton disabled label="Radio label" />);
      expect(wrapper4).toMatchSnapshot();
    });
  });
});
