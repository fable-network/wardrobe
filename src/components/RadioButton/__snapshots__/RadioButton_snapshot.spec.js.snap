// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioButton
    label="Radio label"
    onSelect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          <RadioButton__RadioInput
            onClick={[Function]}
          />
          <RadioButton__StyledRadioInput />
        </div>,
        <RadioButton__Label>
          Radio label
        </RadioButton__Label>,
      ],
      "disabled": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <RadioButton__RadioInput
              onClick={[Function]}
            />,
            <RadioButton__StyledRadioInput />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": undefined,
              "name": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Radio label",
        },
        "ref": null,
        "rendered": "Radio label",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            <RadioButton__RadioInput
              onClick={[Function]}
            />
            <RadioButton__StyledRadioInput />
          </div>,
          <RadioButton__Label>
            Radio label
          </RadioButton__Label>,
        ],
        "disabled": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <RadioButton__RadioInput
                onClick={[Function]}
              />,
              <RadioButton__StyledRadioInput />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "checked": undefined,
                "name": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Radio label",
          },
          "ref": null,
          "rendered": "Radio label",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`RadioButton Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioButton
    label="Radio label"
    onSelect={[Function]}
    selected={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          <RadioButton__RadioInput
            checked={true}
            onClick={[Function]}
          />
          <RadioButton__StyledRadioInput />
        </div>,
        <RadioButton__Label>
          Radio label
        </RadioButton__Label>,
      ],
      "disabled": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <RadioButton__RadioInput
              checked={true}
              onClick={[Function]}
            />,
            <RadioButton__StyledRadioInput />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": true,
              "name": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Radio label",
        },
        "ref": null,
        "rendered": "Radio label",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            <RadioButton__RadioInput
              checked={true}
              onClick={[Function]}
            />
            <RadioButton__StyledRadioInput />
          </div>,
          <RadioButton__Label>
            Radio label
          </RadioButton__Label>,
        ],
        "disabled": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <RadioButton__RadioInput
                checked={true}
                onClick={[Function]}
              />,
              <RadioButton__StyledRadioInput />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "checked": true,
                "name": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Radio label",
          },
          "ref": null,
          "rendered": "Radio label",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`RadioButton Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioButton
    label="Radio label"
    onSelect={[Function]}
    selected={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          <RadioButton__RadioInput
            checked={false}
            onClick={[Function]}
          />
          <RadioButton__StyledRadioInput />
        </div>,
        <RadioButton__Label>
          Radio label
        </RadioButton__Label>,
      ],
      "disabled": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <RadioButton__RadioInput
              checked={false}
              onClick={[Function]}
            />,
            <RadioButton__StyledRadioInput />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "name": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Radio label",
        },
        "ref": null,
        "rendered": "Radio label",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            <RadioButton__RadioInput
              checked={false}
              onClick={[Function]}
            />
            <RadioButton__StyledRadioInput />
          </div>,
          <RadioButton__Label>
            Radio label
          </RadioButton__Label>,
        ],
        "disabled": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <RadioButton__RadioInput
                checked={false}
                onClick={[Function]}
              />,
              <RadioButton__StyledRadioInput />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "checked": false,
                "name": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Radio label",
          },
          "ref": null,
          "rendered": "Radio label",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`RadioButton Component snapshots should match snapshot 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioButton
    disabled={true}
    label="Radio label"
    onSelect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          <RadioButton__RadioInput
            disabled={true}
            onClick={[Function]}
          />
          <RadioButton__StyledRadioInput />
        </div>,
        <RadioButton__Label>
          Radio label
        </RadioButton__Label>,
      ],
      "disabled": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <RadioButton__RadioInput
              disabled={true}
              onClick={[Function]}
            />,
            <RadioButton__StyledRadioInput />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": undefined,
              "disabled": true,
              "name": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Radio label",
        },
        "ref": null,
        "rendered": "Radio label",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            <RadioButton__RadioInput
              disabled={true}
              onClick={[Function]}
            />
            <RadioButton__StyledRadioInput />
          </div>,
          <RadioButton__Label>
            Radio label
          </RadioButton__Label>,
        ],
        "disabled": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <RadioButton__RadioInput
                disabled={true}
                onClick={[Function]}
              />,
              <RadioButton__StyledRadioInput />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "checked": undefined,
                "disabled": true,
                "name": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Radio label",
          },
          "ref": null,
          "rendered": "Radio label",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
